multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/api/impl/OperationsApiServiceImpl.java

```diff
@@ -214,16 +214,33 @@ private Operation getOperation(String operationId) {
             Operation operation = operationService.createQuery()
                                                   .processId(operationId)
                                                   .singleResult();
-            operation = operationsHelper.addState(operation);
+
+            operation = addOperationState(operation);
+
             operation = operationsHelper.addErrorType(operation);
             return operation;
         } catch (NoResultException e) {
             throw new NotFoundException(e, Messages.OPERATION_0_NOT_FOUND, operationId);
         }
     }
 
+    private Operation addOperationState(Operation operation) {
+        long startTime = System.nanoTime();
+        operation = operationsHelper.addState(operation);
+        long totalTimeInNano = System.nanoTime() - startTime;
+        LOGGER.debug("Operation state calculation took {} nanoseconds.", totalTimeInNano);
+        if (operation.getCurrentState() == null) {
+            LOGGER.warn("Cached operation is null for process \"{}\"!", operation.getProcessId());
```

Comment:
Not really sure that `info` is a good choice because the cached state must never be `null`. If it has null values, this means that there is gap in the code somewhere.

Note: Possible case for null value, which is not a gap, will be if there is a deployment during our LSS.

Reply:
Wouldn't it generate a lot of spam at the update of DS when old operations doesn't have value for this column?